src/main/java/org/jabref/logic/pdf/search/retrieval/PdfSearcher.java

```diff
@@ -0,0 +1,64 @@
+package org.jabref.logic.pdf.search.retrieval;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+
+import org.jabref.logic.pdf.search.indexing.EnglishStemAnalyzer;
+import org.jabref.model.pdf.search.PdfSearchResults;
+import org.jabref.model.pdf.search.SearchResult;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.lucene.index.DirectoryReader;
+import org.apache.lucene.queryparser.classic.MultiFieldQueryParser;
+import org.apache.lucene.queryparser.classic.ParseException;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.ScoreDoc;
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.store.SimpleFSDirectory;
+
+import static org.jabref.model.pdf.search.SearchFieldConstants.PDF_FIELDS;
+
+public final class PdfSearcher {
+    private static final Log LOGGER = LogFactory.getLog(PdfSearcher.class);
+
+    private final Directory indexDirectory;
+
+    public PdfSearcher() throws IOException {
+        this.indexDirectory = new SimpleFSDirectory(Paths.get("src/main/resources/luceneIndex"));
+    }
+
+    /**
+     * Search for results matching a query in the Lucene search index
+     *
+     * @param searchString a pattern to search for matching entries in the index, must not be null
+     * @param maxHits      number of maximum search results, must be positive
+     * @return a result set of all documents that have matches in any fields
+     */
+    public PdfSearchResults search(String searchString, int maxHits) throws IOException {
+        if (Objects.requireNonNull(searchString, "The search string was null!").isEmpty()) {
```

Comment:
This is a bit weird, with the isEmpty 
The Objects.requireNonNull will throw an NPE if the field is null. And you can omit the second parameter.
This is a breaking condition. So afterwards you can safely assume that searchString is not null